src/engine/sound/sound_manager.cpp

```diff
@@ -0,0 +1,131 @@
+#include "sound_emitter.h"
+#include "sound_listener.h"
+#include "sound_manager.h"
+
+#include "caudio_include.h"
+
+#include <assert.h>
+#include <iostream>
+
+using namespace thrive;
+
+struct SoundManager::Implementation{
+
+    Implementation(
+        const std::string &device
+    ) : m_initialized(false)
+    {
+
+        // TODO: allow selecting a specific audio device
+
+        m_audioManager = cAudio::createAudioManager(false);
+
+        assert(m_audioManager && "Failed to create audio manager");
+
+		cAudio::IAudioDeviceList* devices = cAudio::createAudioDeviceList();
+
+        // GCC complains about unused variables if devices are not used
+		//auto deviceCount = devices->getDeviceCount();
+        //assert(deviceCount >= 1 && "No audio devices found");
+        
+		auto defaultDeviceName = devices->getDefaultDeviceName();
+
+        const char* selectedDevice = device.empty() ? defaultDeviceName.c_str() : device.c_str();
+        
+        const bool succeeded = m_audioManager->initialize(selectedDevice);
+
+        CAUDIO_DELETE devices;
+		devices = 0;
```

Comment:
why set to zero on a separate line? I don't think integers have copy constructors